src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.cpp

```diff
@@ -63,186 +67,272 @@ bool FlightTaskAutoFollowTarget::activate(const vehicle_local_position_setpoint_
 
+		default:
+			gimbal_height = _home_position_z - _position(2); // Assume target is at home position's altitude
+			break;
+	}
+
+	return gimbal_height;
 }
 
+
 bool FlightTaskAutoFollowTarget::update()
 {
+	// Get the latest target estimator message for target position and velocity
 	_follow_target_estimator_sub.update(&_follow_target_estimator);
+
+	// Debugging uORB message for follow target status
 	follow_target_status_s follow_target_status{};
 
+
 	if (_follow_target_estimator.timestamp > 0 && _follow_target_estimator.valid) {
-		const Vector3f pos_ned_est{_follow_target_estimator.pos_est};
-		const Vector3f vel_ned_est{_follow_target_estimator.vel_est};
-		const Vector3f acc_ned_est{_follow_target_estimator.acc_est};
-		const Vector3f target_position_filtered = calculate_target_position_filtered(pos_ned_est, vel_ned_est, acc_ned_est);
-		const Vector2f offset_vector_filtered = calculate_offset_vector_filtered(vel_ned_est);
-		const Vector3f drone_desired_position = calculate_drone_desired_position(target_position_filtered,
-							offset_vector_filtered);
+		// Update second order target pose filter, with a valid data
+		update_target_pose_filter(_follow_target_estimator);
+
+		const Vector3f target_position_filtered = _target_pose_filter.getState();
+		const Vector3f target_velocity_filtered = _target_pose_filter.getRate();
+
+		// Calculate offset 2D vector to target
+		Vector2f drone_to_target_vector  = Vector2f(target_position_filtered.xy()) - Vector2f(_position.xy());
+		// Calculate heading to the target (for Yaw setpoint)
+		const float drone_to_target_heading = atan2f(drone_to_target_vector(1), drone_to_target_vector(0));
+		// Actual orbit angle measured around the target, which is pointing from target to drone, so M_PI_F difference.
+		const float measured_orbit_angle = matrix::wrap_pi(drone_to_target_heading + M_PI_F);
+
+		// Update the sticks object to fetch recent data
+		_sticks.checkAndUpdateStickInputs();
+		// If valid stick input is available, update follow distance, angle and height via RC commands
+		if(_sticks.isAvailable()) {
+			update_rc_adjusted_follow_height(_follow_height);
+			update_rc_adjusted_follow_distance(_follow_distance, drone_to_target_vector);
+			update_rc_adjusted_follow_angle(_follow_angle_rad, measured_orbit_angle, _tracked_orbit_angle_setpoint_rad);
+		}
+
+		// Update target orientation to track
+		update_target_orientation(target_velocity_filtered.xy(), _tracked_target_orientation_rad);
 
-		// Set position and velocity setpoints
-		float desired_velocity_ff_scale = 0.0f;  // Used to ramp up velocity feedforward, avoiding harsh jumps in the setpoints
+		// Orbital tangential velocity setpoint calculated from orbit angle calculation
+		Vector2f orbit_tangential_velocity{0.0f, 0.0f};
+
+		// Update the new orbit angle (rate constrained)
+		_tracked_orbit_angle_setpoint_rad = update_orbit_angle(orbit_tangential_velocity, _tracked_target_orientation_rad, _tracked_orbit_angle_setpoint_rad);
```

Comment:
@potaito i agree with the nitpick ;) my general perspective is that if it is possible to functionalize an operation with clear simple inputs and outputs, i'd return the new value of e.g. an internal state. this way the function is const, and also easily testable. HOWEVER - i also think it is ok, if the function name explicitly says it is e.g. updating class state X, that the class state X does not need to be input, and can directly be modified. the idea is just to make it very easy to know what the function is doing. and not have side effects (i.e. stuff happening that the function didnt tell you it was going to do). 

>I think I'm just nitpicking because some class members are passed as method arguments, but not all

yeah, I'd also be careful about mixing things like this. can add confusion at times. again - the rule of thumb is generally just whatever is easily readable and not lying to the reader.

Reply:
Added the update trajectory part to the function name to be not deceptive 👅